tracer/test/Datadog.Trace.TestHelpers/MockTracerAgent.cs

```diff
@@ -14,20 +14,20 @@
 using System.Net.Sockets;
 using System.Text;
 using System.Threading;
+using System.Threading.Tasks;
 using Datadog.Trace.ExtensionMethods;
 using Datadog.Trace.HttpOverStreams;
 using Datadog.Trace.Util;
 using MessagePack; // use nuget MessagePack to deserialize
-using Xunit.Abstractions;
 
 namespace Datadog.Trace.TestHelpers
 {
     public class MockTracerAgent : IDisposable
     {
         private readonly HttpListener _listener;
         private readonly UdpClient _udpClient;
-        private readonly Thread _tracesListenerThread;
-        private readonly Thread _statsdThread;
+        private readonly Task _tracesListenerTask;
+        private readonly Task _statsdTask;
```

Comment:
Out of interest, were these changes required for UDS, or just cleanup?

Reply:
I changed `HandleUdsTraces` (called by `_tracesListenerThread`) to async because it was waiting asynchronously on a task and that was triggering my OCDs 😛  Then I also changed `_statsdThread` for consistency